torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -413,11 +413,15 @@ void FunctionParameter::set_default_str(const std::string& str) {
     }
   } else if (type_ == ParameterType::LAYOUT) {
     if (str == "None") {
-      default_layout = nullptr;
+      // Assert if default_layout was initialized by checking if the value is
+      // value.
+      TORCH_INTERNAL_ASSERT_DEBUG_ONLY(default_layout == at::Layout::Strided
+          || default_layout == at::Layout::Sparse
+          || default_layout == at::Layout::Mkldnn);
```

Comment:
@ezyang  Here I just check if the value of the enum is valid. Not sure if this is ok. And not sure if I should assign default_layout a value, like Strided by default, in this case.

Reply:
No, this doesn't make sense. `default_layout` is uninitialized memory at this point in time, you're going to fail ASAN if you do this. I suggest that you should just assert that the argument is "optional" (in which case we will correctly skip looking at `default_layout` if someone tries to grab it) and maybe, for good luck, audit the code gen to make sure that this invariant is respected.